api/resources.go

```diff
@@ -7,11 +7,12 @@ import (
 // Resources encapsulates the required resources of
 // a given task or task group.
 type Resources struct {
-	CPU      *int               `hcl:"cpu,optional"`
-	MemoryMB *int               `mapstructure:"memory" hcl:"memory,optional"`
-	DiskMB   *int               `mapstructure:"disk" hcl:"disk,optional"`
-	Networks []*NetworkResource `hcl:"network,block"`
-	Devices  []*RequestedDevice `hcl:"device,block"`
+	CPU         *int               `hcl:"cpu,optional"`
+	MemoryMB    *int               `mapstructure:"memory" hcl:"memory,optional"`
+	MemoryMaxMB *int               `mapstructure:"memory_max" hcl:"memory_max,optional"`
```

Comment:
In this iteration, I've opted to simply add a `memory_max` field in the job spec, with `memory` remaining as the "reserve"/base memory requirement. Happy to reconsider this and use an alternative name for the "base", e.g. `memory_reserve`,`memory_required`?

I considered `memory_min` - but I find it ambiguous. `min` indicates the minimum memory a task uses rather than how much memory we should reserve/allocate for the task.